File: java/src/main/java/com/pahappa/systems/commssdk/v1/models/ApiResponseCode.java
package com.pahappa.systems.commssdk.v1.models;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

public enum ApiResponseCode {
    OK, Failed;

    @JsonValue
    public String toJson() {
        return name();
    }

    @JsonCreator
    public ApiResponseCode fromJson(String json) {
        for (ApiResponseCode code : ApiResponseCode.values()) {
            if (code.name().equalsIgnoreCase(json)) {
                return code;
            }
        }
        throw new IllegalArgumentException("Unknown value: " + json);
    }
}

File: kotlin/src/main/kotlin/v1/models/ApiResponseCode.kt
package v1.models

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue

enum class ApiResponseCode {
    OK, Failed;

    @JsonValue
    fun toJson(): String {
        return name
    }

    @JsonCreator
    fun fromJson(json: String?): ApiResponseCode {
        for (code in ApiResponseCode.entries) {
            if (code.name.equals(json, ignoreCase = true)) {
                return code
            }
        }
        throw IllegalArgumentException("Unknown value: $json")
    }
}
